tempesta_fw/http_msg.c

```diff
@@ -687,25 +687,58 @@ static int
 __hdr_add(TfwHttpMsg *hm, const TfwStr *hdr, unsigned int hid)
 {
```

Comment:
I propose to get rid of the changes introduced by the commit. It's unclear whether do we need to mangle headers longer than 4KB and this is definitely not in the scope of #1475, but I just caught Oops on this branch:
```
[181764.142253] BUG: unable to handle kernel paging request at ffff9e2c1ae3f840
[181764.145584] IP: tfw_msg_iter_move+0x23/0x130 [tempesta_fw]
[181764.148306] PGD 6c126067 P4D 6c126067 PUD 0
[181764.150899] Oops: 0000 [#1] SMP PTI
[181764.152061] Modules linked in: tempesta_fw(O) tempesta_db(O) tempesta_tls(O) tempesta_lib(O) tcp_diag inet_diag xt_tcpudp nft_counter nft_meta nft_chain_route_ipv4 nf_tables_ipv4 xt_mark nft_compat nf_tables nfnetlink bochs_drm ttm drm_kms_helper crct10dif_pclmul drm ppdev fb_sys_fops syscopyarea intel_rapl_perf sg sysfillrect sysimgblt serio_raw parport_pc parport button binfmt_misc ip_tables x_tables ext4 crc16 mbcache jbd2 fscrypto sr_mod sd_mod cdrom ata_generic virtio_net ata_piix libata psmouse virtio_pci virtio_ring virtio i2c_piix4 scsi_mod [last unloaded: tempesta_lib]
[181764.152061] CPU: 1 PID: 12682 Comm: python2 Tainted: G        W  O    4.14.32-kdump+ #217
[181764.152061] Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 1.13.0-3.fc33 04/01/2014
[181764.152061] task: ffff9e31b2c426c0 task.stack: ffffbc648425c000
[181764.152061] RIP: 0010:tfw_msg_iter_move+0x23/0x130 [tempesta_fw]
[181764.152061] RSP: 0018:ffff9e31bfd035f8 EFLAGS: 00010283
[181764.152061] RAX: ffff9e31b2deecc0 RBX: 0000000000000000 RCX: ffff9e2c1ae3f840
[181764.152061] RDX: 0000000000000000 RSI: ffff9e31bfd03618 RDI: ffff9e31bfd03620
[181764.152061] RBP: ffff9e317dbbd020 R08: ffff9e31b2deeb00 R09: ffff9e31b2deea00
[181764.152061] R10: 000000000000000f R11: ffff9e31bfd23620 R12: ffffffffc05a2927
[181764.152061] R13: ffff9e31bfd037d0 R14: ffff9e31bfd03810 R15: 000000000000000f
[181764.152061] FS:  00007f6e7e0c4740(0000) GS:ffff9e31bfd00000(0000) knlGS:0000000000000000
[181764.152061] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[181764.152061] CR2: ffff9e2c1ae3f840 CR3: 0000000078890003 CR4: 00000000003606e0
[181764.152061] Call Trace:
[181764.152061]  <IRQ>
[181764.152061]  __hdr_add+0x135/0x360 [tempesta_fw]
[181764.152061]  ? _raw_spin_unlock+0x1f/0x30
[181764.152061]  tfw_http_msg_hdr_xfrm_str+0xa2/0x430 [tempesta_fw]
[181764.152061]  ? tfw_tls_hs_over+0x43/0x60 [tempesta_fw]
[181764.152061]  tfw_http_msg_hdr_xfrm+0xaf/0x100 [tempesta_fw]
[181764.152061]  tfw_http_req_cache_cb+0x305/0x480 [tempesta_fw]
[181764.152061]  ? tfw_http_req_cache_cb+0x2d9/0x480 [tempesta_fw]
[181764.152061]  ? tfw_http_conn_repair+0xa40/0xa40 [tempesta_fw]
[181764.152061]  tfw_cache_process+0x6d/0x290 [tempesta_fw]
[181764.152061]  tfw_http_msg_process_generic+0x4e3/0xce0 [tempesta_fw]
[181764.152061]  ? ttls_decrypt+0x31f/0x610 [tempesta_tls]
[181764.152061]  tfw_http_msg_process+0x7f/0xc0 [tempesta_fw]
[181764.152061]  ? ttls_recv+0xad/0x640 [tempesta_tls]
[181764.152061]  __gfsm_fsm_exec+0x51/0x90 [tempesta_fw]
[181764.152061]  tfw_gfsm_move+0x146/0x180 [tempesta_fw]
[181764.152061]  tfw_tls_msg_process+0x334/0x510 [tempesta_fw]
[181764.152061]  __gfsm_fsm_exec+0x51/0x90 [tempesta_fw]
[181764.152061]  ? consume_skb+0x39/0x120
[181764.152061]  tfw_connection_recv+0x3c/0x60 [tempesta_fw]
[181764.152061]  ? tfw_connection_send+0x20/0x20 [tempesta_fw]
[181764.152061]  ss_tcp_process_data+0x1e5/0x480 [tempesta_fw]
[181764.152061]  ss_tcp_data_ready+0x4d/0xa0 [tempesta_fw]
```